sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -69,13 +69,19 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         val table = synchronized {
           client.getTable(in.database, in.name)
         }
-        val schemaString = table.getProperty("spark.sql.sources.schema")
+        val schemaString = Option(table.getProperty("spark.sql.sources.schema.numOfParts")) match {
+          case Some(numOfParts) =>
+            val parts = (0 until numOfParts.toInt).map { index =>
+              Option(table.getProperty(s"spark.sql.sources.schema.part.${index}"))
+                .getOrElse("Could not read schema from the metastore because it is corrupted.")
```

Comment:
should probably throw an exception here instead of returning a string which gets mkstring-ed.